tests/rptest/tests/throughput_limits_snc_test.py

```diff
@@ -204,3 +208,122 @@ def check_node_quota_metric(self, config_prop: self.ConfigProp,
+
+
+class ThroughputLimitsSnc(ThroughputLimitsSncBase):
+    """
+    A generic class for tests for throughput limiting that works at
+    shard/node/cluster (SNC) levels
+    """
+    def __init__(self, test_ctx: TestContext, *args, **kwargs):
+        super(ThroughputLimitsSnc, self).__init__(test_ctx, *args, **kwargs)
+        # at least 3 partitions needed overall to have 3 partition leaders
+        # at the same node,
+        self.topics = (TopicSpec(partition_count=3), )
+        self.failed_checks = []
+        self.total_checks = 0
+
+    def check_equal_epsilon(self, measured: int, expected: int, epsilon: float,
+                            what: str):
+        error = (measured - expected) / expected
+        measured = int(measured)
+        if abs(error) > epsilon:
+            self.failed_checks.append(
+                f"{what} ({measured}) is not within ±{epsilon*100}% error "
+                f"of the expected value ({expected}), "
+                f"actual error: {float(f'{error*100:.4g}'):+g}%")
+            self.logger.error(
+                f"(FAIL) {what}: {measured} == {expected} {float(f'{error:.4g}'):+g} < ±{epsilon*100}%"
+            )
+        else:
+            self.logger.info(
+                f"(PASS) {what}: {measured} == {expected} {float(f'{error:.3g}'):+g} < ±{epsilon*100}%"
+            )
+        self.total_checks += 1
+
+    @cluster(num_nodes=4)
+    @parametrize(config_quota_node_max=8 * 1024 * 1024)
+    @parametrize(config_quota_node_max=2 * 1024 * 1024)
+    @parametrize(config_quota_node_max=512 * 1024)
+    def test_node_limits(self, config_quota_node_max):
+        """
+        Apply various backpressure to one node of the cluster.
+        Verify that node throughput limits are applied.
+        """
+
+        # add partitions to the topic until there are 3 partition
+        # leaders on any of the nodes
+        partitions_cnt = len(self.redpanda.partitions(self.topic))
+        while True:
+            while True:
+                parts = self.redpanda.partitions(self.topic)
+                if len(parts) < partitions_cnt:
+                    self.logger.debug(
+                        f"New partition not arrived yet, waiting. "
+                        f"{parts} < {partitions_cnt}")
+                    continue
+                parts_c = Counter([p.leader for p in parts])
+                if parts_c[None] == 0:
+                    break
+                self.logger.debug(f"Waiting for leader of the new partition")
+                time.sleep(0.1)
+
+            parts_c_m = parts_c.most_common(1)
+            most_active_node, most_active_count = parts_c_m[0]
+            self.logger.debug(
+                f"Most active node is {most_active_node.name} "
+                f"with {most_active_count} leaders. All stat: {parts_c}")
+            if most_active_count == 3:
+                break
+            assert most_active_count < 3
+
+            self.rpk.add_topic_partitions(self.topic, 1)
```

Comment:
Would it make for a better test to perform these topic modifications while a producer is producing? Its likely that will occur in production.

Reply:
I think that is a good idea for future. For now I'm struggling with more basic things in this test.